de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/NegotiationHandler.java

```diff
@@ -0,0 +1,90 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.log4j.Logger;
+import de.fu_berlin.inf.dpp.negotiation.SessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.OutgoingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.ProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.AbstractOutgoingProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.AbstractIncomingProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.NegotiationTools;
+import de.fu_berlin.inf.dpp.server.progress.ConsoleProgressIndicator;
+import de.fu_berlin.inf.dpp.session.INegotiationHandler;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.util.NamedThreadFactory;
+
+public class NegotiationHandler implements INegotiationHandler {
+
+    private static final Logger log = Logger
+        .getLogger(NegotiationHandler.class);
+
+    private final ISarosSessionManager sessionManager;
+    private final ExecutorService executor = Executors.newCachedThreadPool(new NamedThreadFactory(
+        "ServerNegotiationHandler-"));
+
+    public NegotiationHandler(ISarosSessionManager sessionManager) {
+        sessionManager.setNegotiationHandler(this);
+        this.sessionManager = sessionManager;
+    }
+
+    @Override
+    public void handleOutgoingSessionNegotiation(
+        final OutgoingSessionNegotiation negotiation) {
+
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                SessionNegotiation.Status status = negotiation
+                    .start(new ConsoleProgressIndicator(System.out));
+
+                if (status == Status.OK) sessionManager.startSharingProjects(negotiation.getPeer());
+                else if (status == Status.ERROR || status == Status.REMOTE_ERROR) log.error(negotiation.getErrorMessage());
```

Comment:
I do not know what information is contained in the getErrorMessage(). I think it will not hurt to have a better/unique log statement for ERROR,REMOTE_ERROR, LOCAL_CANCEL and REMOTE_CANCEL